src/scout_apm/falcon.py

```diff
@@ -0,0 +1,90 @@
+# coding=utf-8
+from __future__ import absolute_import, division, print_function, unicode_literals
+
+from scout_apm.api import install
+from scout_apm.core.ignore import ignore_path
+from scout_apm.core.tracked_request import TrackedRequest
+
+try:
+    import falcon
+except ImportError:  # pragma: no cover
+    falcon = None
+
+# Falcon Middleware docs:
+# https://falcon.readthedocs.io/en/stable/api/middleware.html
+
+
+class ScoutMiddleware(object):
+    """
+    Falcon Middleware for integration with Scout APMi.
+    """
+
+    def __init__(self, config):
+        install(config=config)
+        self.api = None
+
+    def set_api(self, api):
+        if not isinstance(api, falcon.API):
+            raise ValueError("api should be an instance of falcon.API")
+        self.api = api
+
+    def process_request(self, req, resp):
+        if self.api is None:
+            raise RuntimeError(
```

Comment:
I think historically we have logged failure without raising exceptions so that we never break an application in case of a configuration or unexpected error.

Reply:
Right. I thought this case might be different since it's a missing step in configuration, but I guess it's not, so I'll move to logging an error if `self.api` is `None` and then having an alternative path for naming the operation from just the HTTP method which ignores the suffix and thus which Python method is called i.e. `Controller/tests.integration.test_falcon.HomeResource.on_get` would be just `Controller/tests.integration.test_falcon.HomeResource.GET`.